src/inventory/internal/InventoryRenderer.gd

```diff
@@ -35,11 +34,41 @@ var _inventory: Inventory;
 var _rendering_any_drags: bool;
 
 @onready var _hover := ItemSelection.new(self, _display_hover);
-@onready var _selection := ItemSelection.new(self, _display_selection);
+
+var _selections : Dictionary
 
 func _ready() -> void:
 	mouse_exited.connect(_on_mouse_exited);
 
+func add_selection(id: String, area: Rect2i, fill_color: Color, outline_color: Color) -> bool:
+	var flag: bool = false
+	if !_selections.has(id):
+		_selections[id] = ItemSelection.new(self, _display_selection)
+		flag = true
+	return _selections[id].set_selection(area.position, area.size, fill_color, outline_color) or flag
+
+func get_selection(id: String) -> ItemSelection:
+	if _selections.has(id):
+		return _selections[id]
+	else:
+		return null
```

Comment:
why should this function exist? i can see a case for "is_selection_active" by id, but not sure on letting other things manipulate the item selection.

Reply:
slightly unsafe, but whatever. in future us we trust